matic/src/main/java/network/matic/sdk/Matic.kt

```diff
@@ -41,73 +43,138 @@ class Matic(networkConfig: NetworkConfig) {
   private var watcherApiFactory = WatcherApiFactory(networkConfig)
 
   fun loadERC20Contract(contractAddress: String, parent: Boolean = false): Flowable<ChildERC20> {
-    return Flowable.just(
-      ChildERC20.load(
-        contractAddress,
-        if (parent) web3jParent else web3j,
-        credentials,
-        CustomContractGasProvider()
+    if (isCredentials) {
```

Comment:
Using when instead of if/else looks better. eg -

```
return when(isCredentials){
      true -> Flowable.just(
        ChildERC721.load(
          contractAddress,
          if (parent) web3jParent else web3j,
          credentials,
          CustomContractGasProvider()
        )
      )
      false -> Flowable.just(
        ChildERC721.load(
          contractAddress,
          if (parent) web3jParent else web3j,
          transactionManager,
          CustomContractGasProvider()
        )
      )
    }
```